cfme/tests/cloud_infra_common/test_provisioning.py

```diff
@@ -169,63 +168,55 @@ def test_provision_approval(appliance, provider, vm_name, smtp_test, request,
     # template, host, datastore = map(provisioning.get, ('template', 'host', 'datastore'))
 
     # It will provision two of them
+    # All the subject checks are normalized, because of newlines and capitalization
+
     vm_names = [vm_name + "001", vm_name + "002"]
-    if BZ(1628240).blocks and provider.one_of(CloudProvider):
+    if provider.one_of(CloudProvider):
         requester = ""
+        vm_type = "instance"
     else:
-        requester = "vm_provision@cfmeqe.com "
+        requester = "vm_provision@cfmeqe.com "  # include trailing space for clean formatting
+        vm_type = "virtual machine"
     collection = appliance.provider_based_collection(provider)
-    inst_args = {'catalog': {
-        'vm_name': vm_name,
-        'num_vms': '2'
-    }}
+    inst_args = {
+        'catalog': {
+            'vm_name': vm_name,
+            'num_vms': '2'
+        }
+    }
 
     vm = collection.create(vm_name, provider, form_values=inst_args, wait=False)
     try:
-        if provider.one_of(CloudProvider):
-            vm_type = "instance"
-        else:
-            vm_type = "virtual machine"
-
-        subject = VersionPicker({
-            LOWEST: "your request for a new vms was not autoapproved",
-            "5.10": "your {} request is pending".format(vm_type)
-        }).pick()
-        wait_for(
-            lambda:
-            len(filter(
-                lambda mail:
-                normalize_text(subject) in normalize_text(mail["subject"]),
-                smtp_test.get_emails())) == 1,
-            num_sec=90, delay=5)
-        subject = VersionPicker({
-            LOWEST: "virtual machine request was not approved",
-            "5.10": "{} request from {}pending approval".format(vm_type, requester)
-        }).pick()
-
-        wait_for(
-            lambda:
-            len(filter(
-                lambda mail:
-                normalize_text(subject) in normalize_text(mail["subject"]),
-                smtp_test.get_emails())) == 1,
-            num_sec=90, delay=5)
+        wait_for(lambda: len(smtp_test.get_emails()) >= 2,
+                 num_sec=90,
+                 delay=3)
     except TimedOutError:
-        subjects = ",".join([normalize_text(m["subject"]) for m in smtp_test.get_emails()])
-        logger.error("expected: %s, got emails: %s", subject, subjects)
-        raise
+        pytest.fail('Did not receive at least 2 emails from provisioning request, received: {}'
+                    .format(smtp_test.get_emails()))
+
+    expected_subject = normalize_text("your {} request is pending".format(vm_type))
+    from_subject = normalize_text("{} request from {}pending approval".format(vm_type, requester))
+
+    received_subjects = [normalize_text(m["subject"]) for m in smtp_test.get_emails()]
+    for subject in [expected_subject, from_subject]:
+        soft_assert(any(subject in r_sub for r_sub in received_subjects),
```

Comment:
_optional_ formatting as discussed above

Reply:
Hi @mshriver, these changes look fine to me in general, but I think the expected subject needs to be changed. This is the traceback I received from when I ran this test locally on a 511 appliance - 
```
E           SoftAssertionError: 
E           Expected subject [your virtual machine request is pending], not matched in received subjects [[u'request id 2 your virtual machine request is pending', u'request id 2 virtual machine request from vm provisioncfmeqecom pending approval']] (cfme/tests/cloud_infra_common/test_provisioning.py:204)
E           Expected subject [virtual machine request from vm provisioncfmeqecom pending approval], not matched in received subjects [[u'request id 2 your virtual machine request is pending', u'request id 2 virtual machine request from vm provisioncfmeqecom pending approval']] (cfme/tests/cloud_infra_common/test_provisioning.py:204)
E           Expected subject [your virtual machine request was approved], not matched in received subjects [[u'request id 2 virtual machine request from vm provisioncfmeqecom was approved pending quota validation', u'request id 2 your virtual machine request was approved pending quota validation']] (cfme/tests/cloud_infra_common/test_provisioning.py:248)
E           Expected subject [virtual machine request from vm provisioncfmeqecom was approved], not matched in received subjects [[u'request id 2 virtual machine request from vm provisioncfmeqecom was approved pending quota validation', u'request id 2 your virtual machine request was approved pending quota validation']] (cfme/tests/cloud_infra_common/test_provisioning.py:248)

cfme/fixtures/soft_assert.py:64: SoftAssertionError
```

I think you might want to use different messages depending on the appliance version.

Reply:
Agreed - I noticed that the use of `any()` here is impacting the traceback when the assertion fails.

Reply:
I think the traceback you've included here is from an older commit, before using `any()`. The only PRT failure is in 5.11 (run 33039). I'll still investigate though, thanks @john-dupuy @valaparthvi